### PR TITLE
Attach GA tag to index html

### PR DESCRIPTION
**Pull request by hochiw**
refs oursky/likedao#349

### CODE REVIEW

**Comment by chihimng** [comment]
let's use https://github.com/stafyniaksacha/vite-plugin-radar

**Comment by hochiw** [comment]
this is awkward cos ViteRadar happens during build time so it burns the GA ID into the index.html when it gets generated.

**Comment by chihimng** [comment]
how about:
- https://github.com/react-ga/react-ga
- or [manually inject the script from react (seems tedious)](https://github.com/cychiuae/project-house/commit/03664d27650a752e4113ece399e81802ffa0674f)

**Comment by hochiw** [comment]
@chihimng updated :adore2x: tested by confirming there are analytics request sent every time I click a page

File: react-app/src/navigation/RouterAnalytics.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const location = useLocation();
+  const [isInitialized, setIsInitialized] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (!isInitialized) return;
+
+    ReactGA.pageview(location.pathname + location.search);
+  }, [location, isInitialized]);
+
+  useEffectOnce(
```
Comment (chihimng) [inline react-app/src/navigation/RouterAnalytics.tsx:22]: Let's use useEffect with empty dep array if we are not waiting for first occurrence of an condition

File: react-app/src/navigation/RouterAnalytics.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    () => {
+      if (id) {
+        ReactGA.initialize(id);
```
Comment (chihimng) [inline react-app/src/navigation/RouterAnalytics.tsx:25]: We might want to put this in a provider, so we can add tracking for user actions other than page view e.g. vote, reaction, post comment in the future

**Review by elise-ng — CHANGES_REQUESTED** [review]

File: react-app/src/navigation/RouterAnalytics.tsx (hunk 2 above)
Comment (hochiw) [reply react-app/src/navigation/RouterAnalytics.tsx:25]: 🆗

**Comment by hochiw** [comment]
@chihimng added wallet analytics :adore2x:

**Comment by chihimng** [comment]
please also see if we can track transaction type & set/unset reaction?

**Review by elise-ng — APPROVED** [review]

**Pull request merged**